cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/SubmitApi/Babbage/Transaction.hs

```diff
@@ -130,40 +133,82 @@ hprop_transaction = H.integrationRetryWorkspace 0 "submit-api-babbage-transactio
+        <&> setRequestHeader "Content-Type" ["application/cbor"]
 
-    request <- H.evalM $ parseRequest submitApiRequestEndpoint
-      <&> setRequestBodyFile txbodySignedBinFp
-      <&> setRequestHeader "Content-Type" ["application/cbor"]
+      response <- H.evalM $ httpLbs request
 
-    response <- H.evalM $ httpLbs request
+      getResponseStatusCode response === 202
 
-    getResponseStatusCode response === 202
+    H.byDurationM 5 30 "Expected UTxO found" $ do
+      void $ execCli' execConfig
+        [ "babbage", "query", "utxo"
+        , "--address", Text.unpack $ paymentKeyInfoAddr $ head wallets
+        , "--cardano-mode"
+        , "--testnet-magic", show @Int testnetMagic
+        , "--out-file", work </> "utxo-2.json"
+        ]
 
-  H.byDurationM 5 30 "Expected UTxO found" $ do
-    void $ execCli' execConfig
-      [ "babbage", "query", "utxo"
-      , "--address", Text.unpack $ paymentKeyInfoAddr $ head wallets
-      , "--cardano-mode"
-      , "--testnet-magic", show @Int testnetMagic
-      , "--out-file", work </> "utxo-2.json"
-      ]
+      utxo2Json <- H.leftFailM . H.readJsonFile $ work </> "utxo-2.json"
+      UTxO utxo2 <- H.noteShowM $ H.noteShowM $ decodeEraUTxO sbe utxo2Json
+      txouts2 <- H.noteShow $ L.unCoin . txOutValueLovelace . txOutValue . snd <$> Map.toList utxo2
 
-    utxo2Json <- H.leftFailM . H.readJsonFile $ work </> "utxo-2.json"
-    UTxO utxo2 <- H.noteShowM $ H.noteShowM $ decodeEraUTxO sbe utxo2Json
-    txouts2 <- H.noteShow $ L.unCoin . txOutValueLovelace . txOutValue . snd <$> Map.toList utxo2
+      H.assert $ 5_000_001 `List.elem` txouts2
 
-    H.assert $ 5_000_001 `List.elem` txouts2
+    response <- H.byDurationM 1 5 "Expected UTxO found" $ do
+      txBodySigned <- H.leftFailM $ H.readJsonFile txbodySignedFp
+
+      cborHex <- H.nothingFail $ txBodySigned ^? Aeson.key "cborHex" . Aeson._String
+
+      let txBs = Base16.decodeLenient (Text.encodeUtf8 cborHex)
+
+      H.evalIO $ BS.writeFile txbodySignedBinFp txBs
+
+      let submitApiRequestEndpoint = "POST " <> uriBase <> "/api/submit/tx"
+
+      request <- H.evalM $ parseRequest submitApiRequestEndpoint
+        <&> setRequestBodyFile txbodySignedBinFp
+        <&> setRequestHeader "Content-Type" ["application/cbor"]
+
+      response <- H.evalM $ httpLbs request
+
+      getResponseStatusCode response === 400
+
+      pure response
+
+    H.evalIO $ LBS.writeFile txFailedResponseFp $ redactHashLbs $ getResponseBody response
+
+    H.diffFileVsGoldenFile txFailedResponseFp "test/cardano-testnet-test/files/golden/tx.failed.response.golden"
+
+
+redactHashLbs :: LBS.ByteString -> LBS.ByteString
+redactHashLbs = id
+  . LBS.fromStrict
+  . Text.encodeUtf8
+  . Text.pack
+  . redactHashString
+  . Text.unpack
+  . Text.decodeUtf8
+  . LBS.toStrict
+
+redactHashString :: String -> String
+redactHashString input =
+  subRegex regex input "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
```

Comment:
This is probably fine. You could insert the the known txid into the file during runtime for extra assurance but that's not typical practice for a golden test.